sqld/src/replication/primary/logger.rs

```diff
@@ -790,6 +874,29 @@ impl ReplicationLogger {
     }
 }
 
+fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
+    unsafe {
+        let conn = rusqlite::Connection::open(data_path)?;
+        conn.execute("VACUUM", ())?;
+        let mut num_checkpointed: c_int = 0;
+        let rc = rusqlite::ffi::sqlite3_wal_checkpoint_v2(
+            conn.handle(),
+            std::ptr::null(),
+            SQLITE_CHECKPOINT_TRUNCATE,
+            &mut num_checkpointed as *mut _,
+            std::ptr::null_mut(),
+        );
+
+        // TODO: ensure correct page size
```

Comment:
if you want a quick validation, just call `pragma page_size`; I don't mind if it's only added as a follow-up later